apps/backend/src/auth/InviteAuthorizer.ts

```diff
@@ -0,0 +1,24 @@
+import { inject, injectable } from 'tsyringe';
+
+import { Tokens } from '../config/Tokens';
+import { UserRole, UserWithRole } from '../models/User';
+import { UserAuthorization } from './UserAuthorization';
+
+@injectable()
+export class InviteAuthorization {
+  constructor(
+    @inject(Tokens.UserAuthorization) private userAuth: UserAuthorization
+  ) {}
+
+  public isRoleInviteAuthorized = async (
+    agent: UserWithRole | null,
+    roleIds?: number[]
+  ) => {
+    if (roleIds === undefined || roleIds.length === 0) return true;
+
```

Comment:
Here if there are no roleIds do we want to authorise.

Reply:
Yes, it is logically ok to add 0 or no roles in an invite for everyone

Reply:
If by "use case" you mean if scenarios where role_ids are undefined or empty will happen then yes, as you might create an invite without a user role and in that scenario you would not want the Authorizer to fail.